packages/flutter/lib/src/widgets/scroll_activity.dart

```diff
@@ -527,26 +532,34 @@ class BallisticScrollActivity extends ScrollActivity {
     ScrollActivityDelegate delegate,
     Simulation simulation,
     TickerProvider vsync,
-  ) : super(delegate) {
+  ) : _simulation = simulation, super(delegate) {
     _controller = AnimationController.unbounded(
       debugLabel: kDebugMode ? objectRuntimeType(this, 'BallisticScrollActivity') : null,
       vsync: vsync,
     )
       ..addListener(_tick)
-      ..animateWith(simulation)
+      ..animateWith(_simulation)
        .whenComplete(_end); // won't trigger if we dispose _controller first
   }
 
   late AnimationController _controller;
 
+  final Simulation _simulation;
+
   @override
   void resetActivity() {
     delegate.goBallistic(velocity);
   }
 
   @override
   void applyNewDimensions() {
-    delegate.goBallistic(velocity);
+    final ScrollPosition position = delegate.scrollPosition;
+    final dynamic simulation = _simulation;
+    if (position != null && simulation is ScrollSimulationMixin){
```

Comment:
nit: space between ) and {

Reply:
The `is ScrollSimulationMixin` is also iffy as was already called out in the previous review, I think?

Reply:
This has not been addressed yet, and cannot be accepted in its current state.